OneCommunityGlobal/HGNRest
# Gary add materials page

# Description
This pull request adds the backend logic for the Purchase Request: Material form in the BM Dashboard (Phase 2).

## Related PRS (if any):
This PR must be checked out to test [#1427 frontend PR](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1427)

## Main changes explained:
- Add BM Dashboard collections to dev database: buildingInventoryTypes, buildingMaterials
- Add Mongoose schema files buildingInventoryType, buildingMaterial
- Add GET route to return item type data
- Add POST route to either create a new document in buildingMaterials collection (if the material is being ordered to the project for the first time), or update purchase record sub-document in an existing document (if the material has been ordered previously for the project). In either case, the purchaseRecord sub-document will have default values added to date, requestedBy and status fields, in addition to the submitted form data.

## How to test:
1. `git checkout gary_add_materials_page`
2. `npm install`
3. `npm run dev`
4. check the related frontend PR for further instructions
5. Optional: connect to the dev db via Mongo Compass, open the buildingMaterials collection, and observe that documents and sub-documents are being updated correctly.

## Notes
- **Please remember to either Approve or Request Changes. Do not just leave a comment.**
- **Please leave a review for BOTH front and backend branches after you have finished testing!**
- Gary began work on this branch and has since left the project. The branch and pull request are currently being managed by Tim.
